packages/xrpl/src/models/transactions/common.ts

```diff
@@ -500,7 +505,42 @@ export function validateCredentialsList(
   }
 }
 
-function containsDuplicates(objectList: object[]): boolean {
-  const objSet = new Set(objectList.map((obj) => JSON.stringify(obj)))
-  return objSet.size !== objectList.length
+/**
+ * Check if an array of objects contains any duplicates.
+ *
+ * @param objectList - Array of objects to check for duplicates
+ * @returns True if duplicates exist, false otherwise
+ */
+export function containsDuplicates(
+  objectList: AuthorizeCredential[] | string[],
+): boolean {
+  // Case-1: Process a list of string-IDs
+  if (typeof objectList[0] === 'string') {
+    const objSet = new Set(objectList.map((obj) => JSON.stringify(obj)))
+    return objSet.size !== objectList.length
+  }
+
+  // Case-2: Process a list of nested objects
+  const seen = new Set<string>()
+
+  // Type guard to ensure we're working with AuthorizeCredential[]
+  // Note: This is not a rigorous type-guard. A more thorough solution would be to iterate over the array and check each item.
+  function isAuthorizeCredentialArray(
+    list: AuthorizeCredential[] | string[],
+  ): list is AuthorizeCredential[] {
+    return typeof list[0] !== 'string'
```

Comment:
This is just the "else" case for the if statement on line 518. I think it's fine to just use the else statement with a comment to explain that this is the case for `AuthorizeCredential[]`

Reply:
It's unnecessary to have this inner function.

Reply:
@pdp2121  yes, you are correct. I have included a comment on line 526.

@khancode  I felt the type-guards offered by the inner function are more robust than a usual else-condition statement.

Reply:
It's best practice to avoid inner functions in JS because of how closure works. If you want to keep this helper function, then I would define it separately outside of this function.

Reply:
ok, updated in [2b3580a](https://github.com/XRPLF/xrpl.js/pull/2874/commits/2b3580a276dd73e2f6068397afc6a77ffde6a357)